### DISCUSSION TITLE
Not all subtitles languages in final output

### DISCUSSION THREAD

**Discussion by mpek** [RandomEngy/VidCoder#1068] [Ideas]
Hello, 
problem is: only english subtitles in encoded output. 
I would like to add all available subtitles of two languages to my encodings (german and english), when available in source material. 

Therefore i was chosing desired languages as a default, to let it chose automatically while importing a  file or directory to queue. 

Source material is MKV with DVD PGS subtitles. 

Please have a look at my settings: 
https://i.imgur.com/GsMAV3a.png

Do you know why the available subtitles are not added automatically regarding my setting? Only english was chosen after importing automatically. 

Did I set something wrongly?

I recognized, that Vidcoder is able to be set to watch for "German (Deutsch)" subtitles, but does not seem to recognize them when they are called only "Deutsch" without German in it. Could that be the issue?

**Comment by RandomEngy** [comment]
Subtitle tracks have a special language marker that is separate from the subtitle name. If the source video did not include this marker, the Picker won't choose it to encode.

I added in some logic to search in the track name as well; this will be in the next Beta release.

**Reply by RandomEngy** [reply]
Change is in 8.10 Beta.

**Comment by mpek** [comment]
I recognized as a target mp4 container supports only one PGS subtitle. Soon I will try ssa or srt subtitles to include into mp4 with your changes. Thank you for change and follow up!

**Comment by mpek** [comment]
I found a way to convert PGA subtitles to SRT, that format is compatible with Handbrake/VidCoder. 
VidCoder adds more than one soft subtitle to final output. 
It works with two and even three subtitles, but VidCoder won't pick them up automatically when importing media. 

I opened a feature request to support also batch importing [here](https://github.com/RandomEngy/VidCoder/issues/1075#issue-1432894876).

**Comment by blackfox135** [comment]
I want to convert all my MKV files to MP4 and I want to burn in all ENGLISH SDH subs to ENGLISH audio tracks.  I can do this manually for each selected video, but I want to add this criteria to my picker file so it always burns SDH subs to the video tracks. Thanks

**Reply by RandomEngy** [reply]
That would be possible I think but would only work for tracks that had been explicitly marked with the ClosedCaptions attribute. When you are encoding these manually, do you see them as "English, Closed Caption"?

**Comment by blackfox135** [comment]
No typically they are English SDH or UTF

 

 

Save the Date Sunday, May 28th, 2023!

Sunday, May 28th will be a very special day in Comox for Bev and Paul.

We hope you will be able to join us!

Details to come.   . . . In the meantime, please “Save The Date!!”

Warmest regards,

Paul & Bev

**Reply by RandomEngy** [reply]
That would make it harder to add automation for it; Closed Captions is the only explicit tag that the HandBrake API provides for the subtitle tracks. Is there any reliable way the tracks are named that would indicate that it's a SDH track?

**Reply by mpek** [reply]
If i may add my two cents with this interesting automation topic. 
Typically subtitle tracks don't have a specific mark when they are sdh tracks, they are just called the same as a normal subtitle, by its country language. 

The only way of achieving this would be with external subtitles that have a "sdh" in its name, that Vidcoder could look for. That means you'd need to extract all subtitles, check which are the sdh tracks, rename them and throw them into Vidcoder. This seems very time consuming but doable. You could also download sdh tracks if available and go from there. But at the moment there seems to be no automation possible.

**Comment by blackfox135** [comment]
Thanks!
I'm still experimenting so will let you know my progress.
Paul

sent from my outlawed Chinese mobile phone
